doc: ethernet 3.7 release notes

decsny opened the pull request:
Release notes for 3.7 ethernet drivers.

Commit:
doc: release-notes-3.7: Add ethernet release notes

Add ethernet driver release notes for 3.7 release.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
doc: migration-guide-3.7: Add KSZ8081 DT changes

Add migration guide info for the DT property names of the
microchip,ksz8081 binding being changed.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

aescolar commented on doc/releases/migration-guide-3.7.rst:
```diff
@@ -238,6 +238,10 @@ Device Drivers and Devicetree
 
+ * The DT properties ``mc,interface-type``, ``mc,reset-gpio``, and ``mc,interrupt-gpio`` of
+   the :dtcompatible:`microchip,ksz8081` phy binding have changed to
+   ``microchip,interface-type``, ``reset-gpios``, and ``int-gpios``, respectively.
```
could you refer to the PR that did this change with ```(:github:`12345`)``` ?

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -366,10 +366,67 @@ Drivers and Sensors
 
-  * Deperecated eth_mcux driver in favor of the reworked nxp_enet driver.
-  * Driver nxp_enet is no longer experimental.
+  * Introduced :kconfig:option:`ETH_DRIVER_RAW_MODE`. This option allows building
```
```suggestion
  * Introduced :kconfig:option:`CONFIG_ETH_DRIVER_RAW_MODE`. This option allows building
```

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -366,10 +366,67 @@ Drivers and Sensors
+  * Added support for the realtek RTL8211F phy.
+  * Deperecated eth_mcux driver.
+  * eth_nxp_enet driver is no longer experimental.
+  * Added support for network device power management with nxp_enet driver on kinetis platforms.
```
```suggestion
  * Added support for network device power management with nxp_enet driver on Kinetis platforms.
```

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -366,10 +366,67 @@ Drivers and Sensors
+    generic ethernet L2 code.
+  * Added new eth_nxp_enet_qos driver for the ethernet controller present on NXP MCXN SOCs.
+  * Added support for adin1100 phy.
+  * Added support for the realtek RTL8211F phy.
```
```suggestion
  * Added support for the Realtek RTL8211F phy.
```

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -366,10 +366,67 @@ Drivers and Sensors
+    managed by the kernel rather than a manual infinite loop.
+  * Disabled hardware checksum acceleration when IPV6 is enabled with eth_nxp_enet, since
+    the hardware does not support accelerating ICMPv6 checksums.
+  * Added support for ENET_1G into the nxp_enet driver.
```
```suggestion
  * Added support for :kconfig:option:`CONFIG_ETH_NXP_ENET_1G` into the nxp_enet driver.
```
?

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -366,10 +366,67 @@ Drivers and Sensors
+  * Disabled hardware checksum acceleration when IPV6 is enabled with eth_nxp_enet, since
+    the hardware does not support accelerating ICMPv6 checksums.
+  * Added support for ENET_1G into the nxp_enet driver.
+  * Added support to use a fused mac address for nxp_enet MAC on some platforms.
```
```suggestion
  * Added support to use a fused MAC address for nxp_enet MAC on some platforms.
```

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -366,10 +366,67 @@ Drivers and Sensors
     reworked to use the new :dtcompatible:`nxp,enet` binding.
+  * Implemented/reworked HW MAC Address filtering in the eth_mcux, eth_nxp_enet,
+    and eth_nxp_s32_gmac, eth_stm32, and eth_nxp_s32_netc drivers.
+  * The eth_nxp_s32_gmac driver now implies :kconfig:option:`MDIO`.
```
```suggestion
  * The eth_nxp_s32_gmac driver now implies :kconfig:option:`CONFIG_MDIO`.
```

pdgendt reviewed (commented):
Fix the compliance check, and maybe group items per driver? A lot has to do with NXP ENET for example.

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -366,10 +366,67 @@ Drivers and Sensors
+  * Adjusted eth_adin2111 driver to have the appropriate multicaster filter mask.
+  * Fixed the "generic spi without crc8" mode of adin2111 driver.
+  * Added Open Alliance SPI protocol support to the adin2111 driver.
+  * Added custom driver extension apis for adin2111 driver.
```
```suggestion
  * Added custom driver extension APIs for adin2111 driver.
```

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -366,10 +366,67 @@ Drivers and Sensors
+  * Added custom driver extension apis for adin2111 driver.
+  * Enabled support for promiscuous mode in the adin2111 driver.
+  * Moved OA buffers out of device data of the adin2111 driver to save ~32KB of space
+    when using the generic spi protocol.
```
```suggestion
    when using the generic SPI protocol.
```

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -366,10 +366,67 @@ Drivers and Sensors
+  * Moved OA buffers out of device data of the adin2111 driver to save ~32KB of space
+    when using the generic spi protocol.
+  * Various small changes to adin2111 driver.
+  * Added support to esp32 ethernet driver to set the mac address during runtime.
```
```suggestion
  * Added support to esp32 ethernet driver to set the MAC address during runtime.
```

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -366,10 +366,67 @@ Drivers and Sensors
+    when using the generic spi protocol.
+  * Various small changes to adin2111 driver.
+  * Added support to esp32 ethernet driver to set the mac address during runtime.
+  * Fixed the esp32 driver not initializing as needed because the MDIO and ethernet drivers
```
This sentence seems off, did it intend to say :
"Fixed the esp32 driver not initializing and returning an -EALREADY error, due to the MDIO and ethernet drivers sharing the same clock subsystem"
?

decsny replied:
on second inspection, this bug only appeared post 3.6 so I will just remove this bullet

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -366,10 +366,67 @@ Drivers and Sensors
+  * Fixed issue in the eth_stellaris driver where it was previously not taken into account
+    that the number of interrupts received by the driver may be less than the number of
+    data packets received by the ethernet controller.
+  * Added a devicetree property for the enc28j60 to set the the rx filter.
```
```suggestion
  * Added a devicetree property for the enc28j60 to set the Rx filter.
```

decsny replied:
to stay consistent with the rest of the document I will capitalize the X as well

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -366,10 +366,67 @@ Drivers and Sensors
+    resets, autonegotiation, link detection, and missing/spamming logging messages.
+  * Changed property names of the reset and interrupt gpios in the KSZ8081 DT binding.
+  * Fixed DSA driver for KSZ8xxx to correctly initialize LAN devices.
+  * Changed eth_stm32 to use phy apis to access the phy to avoid collisions when multitasking.
```
```suggestion
  * Changed eth_stm32 to use phy APIs to access the phy to avoid collisions when multitasking.
```

aescolar replied:
missed

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -366,10 +366,67 @@ Drivers and Sensors
+  * Fixed DSA driver for KSZ8xxx to correctly initialize LAN devices.
+  * Changed eth_stm32 to use phy apis to access the phy to avoid collisions when multitasking.
+  * Removed legacy STM32Cube HAL API support for STM32 F4, F7, and H7 series.
+  * Added support for rx/tx timestamping to eth_stm32_hal driver.
```
```suggestion
  * Added support for Rx/Tx timestamping to eth_stm32_hal driver.
```

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -366,10 +366,67 @@ Drivers and Sensors
+  * Removed legacy STM32Cube HAL API support for STM32 F4, F7, and H7 series.
+  * Added support for rx/tx timestamping to eth_stm32_hal driver.
+  * Fixed the wrong register address being used for tail tag enable in ksz8863.
+  * Added ability to set mac address at runtime with eth_liteeth driver.
```
```suggestion
  * Added ability to set MAC address at runtime with eth_liteeth driver.
```

aescolar replied:
missed

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -366,10 +366,67 @@ Drivers and Sensors
+  * Added support for rx/tx timestamping to eth_stm32_hal driver.
+  * Fixed the wrong register address being used for tail tag enable in ksz8863.
+  * Added ability to set mac address at runtime with eth_liteeth driver.
+  * Added conditions to enable ptp_clock driver implementation for native_posix when PTP
```
```suggestion
  * Added conditions to enable ptp_clock driver implementation for the native_posix ethernet driver when PTP
```
?

aescolar reviewed (dismissed):
Thanks @decsny

decsny commented:
> Fix the compliance check, and maybe group items per driver? A lot has to do with NXP ENET for example.

I did try to group them by driver, do you mean to make some kind of header for each one?

pdgendt commented:
> I did try to group them by driver, do you mean to make some kind of header for each one?

Yeah, I meant a subsection.

Commit:
doc: release-notes-3.7: Add MDIO release notes

Add release notes for MDIO drivers.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

pdgendt commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -1314,6 +1381,7 @@ Tests and Samples
     deprecated and removed. The sample has been replaced by the sample ``net/cellular_modem``
     based on the ``MODEM_CELLULAR`` device driver.
 
+<<<<<<< HEAD
```
Some git conflict leftover here.

nashif commented:
@decsny  can you please fix the doc build issues?

Commit:
doc: migration-guide-3.7: NXP ENET MDIO change

Add migration guide info about the unit of
MDIO_NXP_ENET_TIMEOUT config changing.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -504,11 +504,75 @@ Drivers and Sensors
+  * Implemented/reworked HW MAC Address filtering in the eth_mcux, eth_nxp_enet,
+    and eth_nxp_s32_gmac, eth_stm32, and eth_nxp_s32_netc drivers.
+  * New Drivers
+     * Added new eth_nxp_enet_qos driver for the ethernet controller present on NXP MCXN SOCs.
```
Empty line needed above this

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -504,11 +504,75 @@ Drivers and Sensors
+     * Added new eth_nxp_enet_qos driver for the ethernet controller present on NXP MCXN SOCs.
+     * Added support for adin1100 phy.
+     * Added support for the Realtek RTL8211F phy.
+  * NXP ENET driver changes
```
Empty line needed below this

decsny replied:
can you explain to me why this is needed ?

aescolar replied:
So it renders as a nested list just like the other. Isn't that what you were going for?
(you can see the render of this PR here:
https://builds.zephyrproject.io/zephyr/pr/75001/docs/releases/release-notes-3.7.html
)

decsny replied:
nevermind, I think I see why, there is a highlight for some reason with the 3 space indent in zephyr docs

decsny replied:
woops, your comment didn't load for me until now

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -504,11 +504,75 @@ Drivers and Sensors
+     * Fixed cache maintain being enabled when using a noncache DMA buffer in nxp_enet driver.
+     * Added MMIO mappings to nxp_enet driver.
+     * Clarified DSA supported with eth_nxp_enet.
+  * NXP S32 ethernet changes
```
same

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -504,11 +504,75 @@ Drivers and Sensors
+  * NXP S32 ethernet changes
+     * The eth_nxp_s32_gmac driver now implies :kconfig:option:`CONFIG_MDIO`.
+     * eth_nxp_s32_netc driver updated to use new MBOX API.
+  * Adin2111 driver changes
```
same

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -504,11 +504,75 @@ Drivers and Sensors
+       when using the generic SPI protocol.
+     * Fixed a build warning in eth_adin2111 driver on 64-bit platforms.
+     * Various small changes to adin2111 driver.
+  * STM32 ethernet driver changes
```
same

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -504,11 +504,75 @@ Drivers and Sensors
+     * Changed eth_stm32 to use phy apis to access the phy to avoid collisions when multitasking.
+     * Removed legacy STM32Cube HAL API support for STM32 F4, F7, and H7 series.
+     * Added support for RX/TX timestamping to eth_stm32_hal driver.
+  * ESP32 ethernet driver changes
```
same

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -504,11 +504,75 @@ Drivers and Sensors
+     * Added support to esp32 ethernet driver to set the MAC address during runtime.
+     * Updated esp32 ethernet driver to work with the version 5.1 of hal_espressif.
+     * Fixed build of esp32 ethernet driver when :kconfig:option:`CONFIG_NET_STATISTICS` is enabled.
+  * Other ethernet driver changes
```
same

aescolar commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -504,11 +504,75 @@ Drivers and Sensors
+       ethernet driver when PTP subsystem is enabled.
+     * Fixed DSA driver for KSZ8xxx to correctly initialize LAN devices.
+     * Fixed the wrong register address being used for tail tag enable in ksz8863.
+  * Phy driver changes
```
same

aescolar commented on doc/releases/migration-guide-3.7.rst:
```diff
@@ -643,6 +647,12 @@ LED Strip
+MDIO
+====
+
+* :kconfig:option:`CONFIG_MDIO_NXP_ENET_TIMEOUT` is now in units of
```
Please add a reference to the github issue that introduced the change ```(:github:`12345`)```

decsny replied:
hmm, I thought I did this, don't know where it went, I'll try again

aescolar commented:
@decsny please note you have missed some of the requested changes.

A separate suggestion regarding the way of working: If when a rebase is needed, the rebase is pushed first, and after the relevant updates to the PR content are pushed, reviewers can see easier from the diffs what has changed that is meaningful.
